internal/mode/static/manager.go

```diff
@@ -265,22 +265,28 @@ func registerControllers(
 		{
 			objectType: &gatewayv1.GatewayClass{},
 			options: []controller.Option{
+				controller.WithK8sPredicate(k8spredicate.GenerationChangedPredicate{}),
```

Comment:
The controller options logic will also have to be updated to support setting multiple predicates: https://github.com/nginxinc/nginx-gateway-fabric/blob/7639fe5bc8e8d4af3cc0778887042e1c887cfa90/internal/framework/controller/register.go#L44

Alternatively, you could use the [And](https://github.com/kubernetes-sigs/controller-runtime/blob/7f316f1309b11c24aa8a96750e694f897555d348/pkg/predicate/predicate.go#L236C1-L236C1) controller-runtime function to create a composite predicate.

Reply:
Updated https://github.com/nginxinc/nginx-gateway-fabric/pull/1422/commits/cc429b2a5b6342052cc2ed2971b3230cbf17df77. I have chosen to use the And operator to create a composite predicate, to minimize the code changes required for this PR.